lib/bap_llvm/llvm_macho_loader.hpp

```diff
@@ -152,6 +137,38 @@ MACHO_SECTION_FIELD(reserved2, uint32_t)
 MACHO_SECTION_FIELD(addr, uint64_t)
 MACHO_SECTION_FIELD(offset, uint32_t)
 
+void entry_point(const macho &obj, ogre_doc &s) {
+    for (auto info : macho_commands(obj)) {
+        if (info.C.cmd == MachO::LoadCommandType::LC_MAIN) {
+            const MachO::entry_point_command *entry_cmd =
+                reinterpret_cast<const MachO::entry_point_command*>(info.Ptr);
+            uint64_t entry = entry_cmd->entryoff;
+            if (obj.isLittleEndian() != sys::IsLittleEndianHost)
+                sys::swapByteOrder(entry);
+            s.entry("entry") << entry;
+            return;
+        }
+    }
+    std::vector<uint64_t> addrs;
```

Comment:
Big binaries can have quite a few section, so I don't think that we should use such inefficient algorithm. You can use `std::optional` if you're afraid of using explicit sentinel values, such as `max_int`.

Reply:
it looks like that `optional` was added only in std+17, so you can't use it. So either use a sentinel or a boolean flag.